drivers/hwinfo/hwinfo_nrf.c

```diff
@@ -58,22 +58,56 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 	if (reason & NRFX_RESET_REASON_DIF_MASK) {
 		flags |= RESET_DEBUG;
 	}
+	if (reason & NRFX_RESET_REASON_SREQ_MASK) {
+		flags |= RESET_SOFTWARE;
+	}
 
-#if !NRF_POWER_HAS_RESETREAS
+#if NRFX_RESET_REASON_HAS_CTRLAP
 	if (reason & NRFX_RESET_REASON_CTRLAP_MASK) {
 		flags |= RESET_DEBUG;
 	}
-	if (reason & NRFX_RESET_REASON_DOG0_MASK) {
-		flags |= RESET_WATCHDOG;
+#endif
+#if NRFX_RESET_REASON_HAS_LPCOMP
+	if (reason & NRFX_RESET_REASON_LPCOMP_MASK) {
+		flags |= RESET_LOW_POWER_WAKE;
+	}
+#endif
+#if NRFX_RESET_REASON_HAS_NFC
+	if (reason & NRFX_RESET_REASON_NFC_MASK) {
+		flags |= RESET_LOW_POWER_WAKE;
+	}
+#endif
+#if NRFX_RESET_REASON_HAS_VBUS
+	if (reason & NRFX_RESET_REASON_VBUS_MASK) {
+		flags |= RESET_POR;
+	}
+#endif
+#if NRFX_RESET_REASON_HAS_CTRLAPSOFT
+	if (reason & NRFX_RESET_REASON_CTRLAPSOFT_MASK) {
+		flags |= RESET_DEBUG;
+	}
+#endif
+#if NRFX_RESET_REASON_HAS_CTRLAPHARD
+	if (reason & NRFX_RESET_REASON_CTRLAPHARD_MASK) {
+		flags |= RESET_DEBUG;
+	}
+#endif
+#if NRFX_RESET_REASON_HAS_CTRLAPPIN
+	if (reason & NRFX_RESET_REASON_CTRLAPPIN_MASK) {
+		flags |= RESET_DEBUG;
 	}
+#endif
+#if !NRF_POWER_HAS_RESETREAS
 	if (reason & NRFX_RESET_REASON_DOG1_MASK) {
 		flags |= RESET_WATCHDOG;
 	}
-	if (reason & NRFX_RESETREAS_SREQ_MASK) {
-		flags |= RESET_SOFTWARE;
```

Comment:
But SREQ is an arm bitvalue, why is it being removed?

Reply:
SREQ was moved to line 61